### PR TITLE
Code coverage: Expense fields changes - 6

### PR DESCRIPTION
**Pull request by Dimple16**
Screenshot 2023-01-19 at 3 19 14 PM

### CODE REVIEW

File: src/app/core/services/expense-fields.service.spec.ts [hunk 1]
```diff
@@ -54,4 +60,28 @@ describe('ExpenseFieldsService', () => {
...
+  });
+
+  it('getAllMap(): should get all expense fields map', (done) => {
+    authService.getEou.and.returnValue(new Promise((resolve) => resolve(extendedOrgUserResponse)));
+    apiService.get.and.returnValue(of(expenseFieldWithSeq));
+
+    expenseFieldsService.getAllMap().subscribe((expenseFieldsMap) => {
+      expect(expenseFieldsMap).toEqual(expenseFieldsMapResponse2);
+    });
+    done();
```
Comment (rvab) [inline src/app/core/services/expense-fields.service.spec.ts:76]: this should inside subscribe no?

File: src/app/core/services/expense-fields.service.spec.ts [hunk 2]
```diff
@@ -54,4 +60,28 @@ describe('ExpenseFieldsService', () => {
...
+  });
+
+  it('filterbyCategory(): should filter expense fields by category', () => {
+    const fields = ['purpose', 'txn_dt', 'vendor_id', 'org_category_id'];
+    expenseFieldsService
+      .filterByOrgCategoryId(expenseFieldsMapResponse, fields, orgCategoryData)
+      .subscribe((expenseFields) => {
+        expect(expenseFields).toEqual(expenseFieldObjData);
```
Comment (rvab) [inline src/app/core/services/expense-fields.service.spec.ts:84]: `done` is not called?

**Review by rvab — CHANGES_REQUESTED** [review]

File: src/app/core/services/expense-fields.service.ts [hunk 3]
```diff
@@ -26,7 +28,8 @@ export class ExpenseFieldsService {
             is_custom: false,
           },
         })
-      )
+      ),
+      map((res) => this.dateService.fixDates(res))
```
Comment (rvab) [inline src/app/core/services/expense-fields.service.ts:32]: Why is this added?

File: src/app/core/services/expense-fields.service.ts [hunk 4]
```diff
@@ -68,7 +71,7 @@ export class ExpenseFieldsService {
     tfcMap: Partial<ExpenseFieldsMap>,
     fields: string[],
     orgCategory: OrgCategory
-  ): Observable<Partial<ExpenseFieldsMap>> {
+  ): Observable<Partial<ExpenseFieldsMap | ExpenseFieldsObj>> {
```
Comment (rvab) [inline src/app/core/services/expense-fields.service.ts:74]: Why create a new model?

File: src/app/core/services/expense-fields.service.ts [hunk 5]
```diff
@@ -113,7 +116,9 @@ export class ExpenseFieldsService {
     );
   }
 
-  getDefaultTxnFieldValues(txnFields: Partial<ExpenseFieldsMap>): DefaultTxnFieldValues {
+  getDefaultTxnFieldValues(
+    txnFields: Partial<ExpenseFieldsMap> | Partial<ExpenseFieldsObj>
```
Comment (rvab) [inline src/app/core/services/expense-fields.service.ts:120]: same as above

**Review by rvab — APPROVED** [review]
Minor comments

File: src/app/core/services/expense-fields.service.ts (hunk 3 above)
Comment (Dimple16) [reply src/app/core/services/expense-fields.service.ts:32]: since the service is not handling the conversion of date to string, we need this as the DB and model type is `Date`, but the value is of string,
